### ISSUE TITLE
Unable to open terminal on Windows 7 Professional SP1 x64, with suggested error message update

### ISSUE THREAD

**Issue by Pysis868** [zaggino/brackets-git#1123]
##### Brackets
Release 1.4 build 1.4.0-16380 (release 273804eb3) 
build timestamp: Mon Jul 27 2015 07:23:52 GMT+0100
##### Git for Windows (msysgit)
Git-1.8.5.2-preview20131230
##### Brackets Git
0.14.28

---

I have a similar error presented in the console that others have had, for similar circumstances or not, in the following issues:
* #850 "Some examples to configure various terminals?"
* #1096 "Cannot enter remote login credentials"
* #965 "undefined"

Here is the error I am experiencing:
```
Brackets Git encountered an error...
ExpectedError: Terminal configuration invalid, restoring defaults. Restart Brackets to apply.
```

I am trying to open the terminal for the first time and am receiving this error in the console with the default settings:
```
[brackets-git] ExpectedError
  at Object.<anonymous> (file:///C:/Users/***/AppData/Roaming/Brackets/extensions/user/zaggino.brackets-git/src/ExpectedError.js:11:31)
  at Object.context.execCb (file:///D:/JStuff/Apps/Brackets/www/thirdparty/requirejs/require.js:1658:33)
  at Object.Module.check (file:///D:/JStuff/Apps/Brackets/www/thirdparty/requirejs/require.js:869:55)
  at Object.Module.enable (file:///D:/JStuff/Apps/Brackets/www/thirdparty/requirejs/require.js:1151:22)
  at Object.Module.init (file:///D:/JStuff/Apps/Brackets/www/thirdparty/requirejs/require.js:782:26)
  at callGetModule (file:///D:/JStuff/Apps/Brackets/www/thirdparty/requirejs/require.js:1178:63)
  at Object.context.completeLoad (file:///D:/JStuff/Apps/Brackets/www/thirdparty/requirejs/require.js:1552:21)
  at HTMLScriptElement.context.onScriptLoad (file:///D:/JStuff/Apps/Brackets/www/thirdparty/requirejs/require.js:1679:29)Utils.js:344 consoleLogErrorHandler.js:89 exports.logErrorErrorHandler.js:97 exports.showErrorTerminal.js:166 (anonymous function)bluebird.js:4988 tryCatch1bluebird.js:2333 Promise$_callHandlerbluebird.js:2349 Promise$_settlePromiseFromHandlerbluebird.js:2521 Promise$_settlePromiseAtbluebird.js:2658 Promise$_rejectPromisesbluebird.js:157 Async$_consumeFunctionBufferbluebird.js:120 Async$consumeFunctionBufferbluebird.js:4260 Promise$_Scheduler
```
I found that is the error I have when the extension cannot find an executable to use, either through the current preference, or the defaults it has hard-coded in.

I then enter in an executable in the form of `...\Git\bin\sh.exe` with the arguments `-c "cd '$1'; bash"` (yes, this is probably weird, which is why I want #850 to be completed :) )

Now when I try to open the terminal with these settings, the code reaches past the `setup()` call, farther than the previous error, `configuredOk` is true, so it calls `open()`.
Then it goes all the way down to the WebSocket's `this._ws.send` command, and just returns from there without anything appearing.

**Comment by zaggino** [comment]
Hi, I have never managed to execute bash directly, but it works fine using the default script in `C:\Program Files (x86)\Git\Git Bash.vbs` and passing a directory as an argument. There's some WScript stuff going on inside it.

**Comment by zaggino** [comment]
Feel free to open a PR if you'll find a solution for your installation, but I don't plan on fixing it myself as it works fine on my PCs both home and at work and I have never had issues with this.

**Comment by Pysis868** [comment]
I guess we'll never know why nothing appeared once we sent that command over through Node unless you know how to debug that part; maybe it needs a wrapper application that creates a window?
Anyway, I did find that .vbs file on my system, and it is working now.

Could the error message be improved to show all of the paths that were tried, with or without the arguments as part of the message?

Then I would just advocate for a more apparent solution according to #850 rather than just relying on the default settings for the system to be placed into the dialog.  Maybe making a mention in the readme too where it shows the recommended settings for Window for example, but instead of just showing the recommended msysgit settings, it would also show the recommended setting for Brackets-Git on Windows.

**Comment by typo3ua** [comment]
Hello!

I do not found file in `C:\Program Files (x86)\Git\Git Bash.vbs`

Where do I find this file?

**Comment by zaggino** [comment]
Hi @typo3ua , install latest Git from https://git-scm.com/downloads
Extension will launch `C:\Program Files\Git\git-bash.exe` by default

**Comment by JustAGuyTryingToCodeSomething** [comment]
This issue started appearing after I rebuilt my Windows development PC. I think it happens because I installed GIT to the D:\ drive. Amending the configuration settings should resolve the issue. See below.

Replacing the Git Settings > Git Configuration > Custom Terminal from  C:\Program Files (x86)\Git\Git Bash.vbs to D:\Apps\Git\git-bash.exe (where Git is installed on my unit) nearly fixed it, but the window opened and closed with an error I couldn't see, so then I removed the $1 from the command arguments input and it works, opening git-bash in current project folder.
